### ISSUE TITLE
[feature] Command to view information set in package_info

### ISSUE THREAD

**Issue by lieser** [conan-io/conan#9031]
For investigating problems, it can be helpful to view the `cpp_info`/`env_info` information that a package in the local cache has.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Definitely. We are trying to not add things to Conan 1.X releases that are not extremely necessary to prepare the path for Conan 2.0, so it seems this will be more a Conan 2 feature. What we are doing right now already in Conan 1.X is improving all the architecture of the ``cpp_info`` and ``env`` and how to access dependencies information.

In any case, take into account that this is a ``conan info`` feature, it is not possible to evaluate the ``cpp_info`` of a single package, because it might depend on the configuration, so getting the ``cpp_info`` will require all settings, options... that is a full profile, and maybe even dependencies.

**Comment by lieser** [comment]
More a nice to have, so completely fine to leave it for 2.0 from my POV.

Regarding the configuration, maybe allow to specify for `conan info` a full reference similar to `conan get zlib/1.2.11@:2144f833c251030c3cfd61c4354ae0e38607a909`.

**Comment by memsharded** [comment]
> Regarding the configuration, maybe allow to specify for conan info a full reference similar to conan get zlib/1.2.11@:2144f833c251030c3cfd61c4354ae0e38607a909.

That is the problem, if you have (which is very typical), something like:

```python
def package_info(self):
      if self.settings.os == "Windows":
             self.cpp_info.libs = ["somewinlib"]
      else:
             self.cpp_info.libs = ["otherlib"]
```

Then you cannot evaluate that based on the package_id, which is a one way hash, impossible to know the settings, options, etc. This is the reason you need a full profile. There are also recipes that might provide different ``package_info()`` based on dependencies, it is not common, but it is possible, so also evaluating the dependencies might be necessary.

**Comment by memsharded** [comment]
The approach for this in 2.0 will be:
- ``conan list pkg/version:package_id`` will display the information from ``info`` summarized in ``conaninfo.txt``, for that specific package (settings, options)
- ``conan graph info`` will display graph information, including ``cpp_info`` information. It is possible to filter the packages displayed with ``--package-filter``

Still, displaying the ``cpp_info`` for a given package from a ``package_id`` is impossible. As the ``package_info()`` cannot be evaluated without full profile and dependencies information. So closing this as solved, Conan 2.0 is displaying the most it can for every case.
